--- Makefile.ORIG	2013-01-28 15:58:23.167041138 +0000
+++ Makefile	2013-01-28 16:18:56.635534341 +0000
@@ -3,7 +3,7 @@
 
 INCBASE = $(X11RDPBASE)/include
 LIBBASE = $(X11RDPBASE)/lib
-
+INSTALL = /usr/bin/install -c
 XSRCBASE = ../build_dir/xorg-server-1.9.3
 
 OBJS = rdpmain.o rdpdraw.o rdpinput.o rdpmisc.o rdpup.o rdprandr.o \
@@ -109,3 +109,6 @@
 
 fbcmap_mi.o: ../build_dir/xorg-server-1.9.3/fb/fbcmap_mi.c
 	$(CC) $(CFLAGS) -c ../build_dir/xorg-server-1.9.3/fb/fbcmap_mi.c
+
+install: all
+	$(INSTALL) X11rdp $(X11RDPBASE)/bin/X11rdp
