--- buildx.sh.orig	2013-01-31 14:32:35.245173991 +0000
+++ buildx.sh	2013-01-31 16:03:11.876826277 +0000
@@ -27,6 +27,15 @@
 {
     file=$1
 
+    #Scarygliders patch to obtain package version # and the name without version #...
+    packageVersion=$(echo $file | grep -Eo '[0-9]+\.[0-9].[0-9]+')
+    if [ "$packageVersion" = "" ]
+    then
+        packageVersion=$(echo $file | grep -Eo '[0-9]+\.[0-9]+')
+    fi
+    basicName=${file%-*}
+
+
     # if we already have the file, don't download it
        if [ -r downloads/$file ]; then
          return 0
@@ -232,7 +241,7 @@
 
     # make module
     if [ ! -e cookies/$mod_name.made ]; then
-        (cd build_dir/$mod_name ; make)
+        (cd build_dir/$mod_name ; make -j 1)
         if [ $? -ne 0 ]; then
             echo ""
             echo "make failed for module $mod_name"
@@ -243,7 +252,7 @@
     fi
 
     # install module
-    (cd build_dir/$mod_name ; make install)
+    (cd build_dir/$mod_name ; checkinstall -D --dpkgflags=--force-overwrite --fstrans=no --pkgversion=$packageVersion --pakdir ./../../../../../packages/Xorg/ --pkgname=x11rdp-$basicName --install=yes --default make install)
     if [ $? -ne 0 ]; then
         echo ""
         echo "make install failed for module $mod_name"
@@ -375,15 +384,15 @@
 
 cd rdp
 make
-if [ $? -ne 0 ]; then
-    echo "error building rdp"
-    exit 1
-fi
 
 # this will copy the build X server with the other X server binaries
 strip X11rdp
 cp X11rdp $X11RDPBASE/bin
 
+# Scarygliders patch to have checkinstall make an X11rdp package file...
+x11rdpVer=$(grep "X11RDPVER" rdp.h | cut -d" " -f3 | grep  -Eo '[0-9]+\.[0-9].[0-9]+')
+checkinstall -D --dpkgflags=--force-overwrite --fstrans=no --pakdir ../../../../packages --pkgname=X11rdp --pkgversion=$x11rdpVer --install=yes --default make install
+
 if [ "$2" = "drop" ]; then
   echo ""
   echo "dropping you in dir, type exit to get out"
